agent/tool-scripts/postprocess/pidstat-postprocess

```diff
@@ -70,88 +69,69 @@ if ( (($? >> 8) == 0) && (! -e "/usr/bin/vdsClient") && (-e "/usr/bin/virsh") )
 	}
 }
 
-# read the pidstat.txt and collect stats for all non dm-* devices
-open(PIDSTAT_TXT, "$dir/pidstat-stdout.txt") || die "could not find $dir/pidstat-stdout.txt\n";
-while ($line = <PIDSTAT_TXT>) {
-	chomp $line;
-	        ##      Time   UID       PID    %usr   %system  %guest  %CPU   CPU     minflt/s     majflt/s     VSZ     RSS       %MEM      kB_rd/s      kB_wr/s    kB_ccwr/s iodelay     cswch/s    nvcswch/s  Command
-	        # 1394568782   107    222936     0.67         0.00        0.00   3       0.67        8       0.00         0.00        164   17760     0.61         0.00        0.00         0.00         0       64.67        0.00      /usr/libexec/qemu-kvm -name dhcp31-2
-	if ( $line =~ /^(\s\d+\s+\d+\s+\d+\s+\d*).*/ ) {
-		$line =~ s/^\s+//;
-		@stats = split(/\s+/,$line);
-		my $timestamp = shift(@stats);
-		my $uid = shift(@stats);
-		my $pid = shift(@stats);
-
-		my $usr = shift(@stats);
-		# $usr might actually be the TID if its an integer
-		if ($usr =~ /^\d+$/) {
-			$tid=$usr;
-			# and the next value is for $usr
-			$usr = shift(@stats);
-		}
-		my $sys = shift(@stats);
-		my $guest = shift(@stats);
-		my $cpu = shift(@stats);
-		my $cpu_id = shift(@stats);
-		my $minor_faults = shift(@stats);
-		my $major_faults = shift(@stats);
-		my $vsz = shift(@stats);
-		my $rss = shift(@stats);
-		my $mem = shift(@stats);
-		my $kb_read = shift(@stats);
-		my $kb_write = shift(@stats);
-		my $kb_ccrw = shift(@stats);
-		my $io_delay = shift(@stats);
-		my $cswitches = shift(@stats);
-		my $nvcswitches = shift(@stats);
-		my $cmd = join(' ', @stats);
-
-		# print "timestamp[$timestamp]  uid[$uid]\n";
-
-		my $timestamp_ms = 1000 * $timestamp;
-
-		# if threads are reported
-		if ( $pid =~ /(\d+)\s+(\d+)/ ) {
-			$pid = $1;
-			$tid = $2;
-			if ($tid == 0) {
-				next;
-			} else {
-				$pid = $tid;
-				$cmd =~ s/\|_+//;
+my %timestamps;
+my @attributes = qw(usr system guest CPU_PCT CPUID minflt_s majflt_s VSZ RSS MEM_PCT kB_rd_s kB_wr_s iodelay cswch_s nvcswch_s);
```

Comment:
Is there a way we can keep this list of attributes stored either here or in `pidstat-convert` to avoid the possibility drift in the future?